### PR TITLE
docs(world): add namespaces-access-control to the new docs

### PR DESCRIPTION
**Pull request by qbzzt**

### CODE REVIEW

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 1]
```diff
@@ -1 +1,74 @@
 # Namespaces and Access Control
+
+MUD access control is based on namespaces.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:3]:
```suggestion
Access control in a `World` is based on namespaces.
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 2]
```diff
@@ -1 +1,74 @@
...
+Every resource is part of a namespace (if there is no namespace specified that just mean that the resource is part of the root namespace).
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:4]:
```suggestion
Every resource is part of a namespace (a blank namespace refers to the `root` namespace).
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 3]
```diff
@@ -1 +1,74 @@
...
+Access to every resource is managed by the owner address of the namespace.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:5]:
```suggestion
Access to resources is managed by the owner of the namespace.
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 4]
```diff
@@ -1 +1,74 @@
...
+
+## Resource identifiers
+
+A `ResourceId` is a 32-byte value that uniquely identifies a resource in a MUD `World`.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:9]:
```suggestion
A `ResourceId` is a 32-byte value that uniquely identifies a resource in a `World`.
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 5]
```diff
@@ -1 +1,74 @@
...
+It is two bytes of resource type followed by 14 bytes of namespace and then 16 bytes of the name of the actual resource.
+
+Currently, MUD supports these resource types:
+
+- [Table (`tb`)](https://github.com/latticexyz/mud/blob/main/packages/store/src/storeResourceTypes.sol#L11).
+  An onchain table whose information that is available both onchain (through calls to `view` functions) and offchain (either through calls to `view` functions, events, or an [indexer](/services/indexer)).
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:15]:
```suggestion
  An onchain table whose information is available both onchain (through calls to `view` functions) and offchain (either through calls to `view` functions, events, or an [indexer](/services/indexer)).
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 6]
```diff
@@ -1 +1,74 @@
...
+- [Offchain table (`ot`)](https://github.com/latticexyz/mud/blob/main/packages/store/src/storeResourceTypes.sol#L14).
+  An offchain table whose information is only available offchain (either through events, or through an [indexer](/services/indexer)).
+- [Namespace (`ns`)](https://github.com/latticexyz/mud/blob/main/packages/world/src/worldResourceTypes.sol#L15).
+  A namespace is a container for tables, offchain tables, and systems, as explained here.
+- [System (`sy`)](https://github.com/latticexyz/mud/blob/main/packages/world/src/worldResourceTypes.sol#L25).
+  A system contains functions that interact with table data onchain.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:21]:
```suggestion
  A [system](./systems) contains logic that interact with table data onchain.
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 7]
```diff
@@ -1 +1,74 @@
...
+- [Module (`md`)](https://github.com/latticexyz/mud/blob/main/packages/world/src/worldResourceTypes.sol#L21).
+  A module is a script that can install tables, systems, etc.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:23]:
```suggestion
  A module is a deployed script that can install tables, systems, etc.
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 8]
```diff
@@ -1 +1,74 @@
...
+
+There are three access levels in MUD.
+
+### "No" access
+
+This is the level that most users of a MUD application have for all resources.
+Users on this level are allowed to do the following:
+
+- Read information (table content).
+  _There are no secrets on the blockchain_, transactions and events are public information, so this information has to be available.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:37]: > There are no secrets on the blockchain

This point is arguable (zk, hash commitments, etc), would remove it here

```suggestion
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 9]
```diff
@@ -1 +1,74 @@
...
+## Access control levels
+
+There are three access levels in MUD.
+
+### "No" access
+
+This is the level that most users of a MUD application have for all resources.
+Users on this level are allowed to do the following:
+
+- Read information (table content).
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:36]:
```suggestion
- Read data from tables.
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 10]
```diff
@@ -1 +1,74 @@
...
+  _There are no secrets on the blockchain_, transactions and events are public information, so this information has to be available.
+- Call functions on any `System` that is publicly accessible to make transactions.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:38]:
```suggestion
- Call functions on [public `System`s](./systems).
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 11]
```diff
@@ -1 +1,74 @@
...
+
+### Access
+
+This level of access is often required for software that is part of the application.
+Access can be granted either on a namespace basis (access to the namespace gives access to all the resources inside it) or on an individual resource.
+
+Entities with this level of access can do anything that "no" access allows, and also:
+
+- Write directly to tables.
+  This is as opposed to users with no access, who can only modify table information by going through a `System` (by default, a `System` has access to its own namespace, which of course includes all of the namespace's tables).
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:48]:
```suggestion
  This is as opposed to users with no access, who can only modify table information by going through a public `System` (by default, a `System` has access to its own namespace, which includes all of the namespace's tables).
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 12]
```diff
@@ -1 +1,74 @@
...
+- Call a `System` that is _not_ publicly accessible.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:49]:
```suggestion
- Call functions on [private `System`s](./systems).
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 13]
```diff
@@ -1 +1,74 @@
...
+- Withdraw from the ETH balance of the namespace.
+  To avoid giving this permission, give an address access to every resource in the namespace, rather than to the namespace itself.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:51]: Can we rephrase this slightly to make it clear in the first sentence that this is only true for access to a `namespace` resource?

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 14]
```diff
@@ -1 +1,74 @@
...
+
+### Ownership
+
+Namespaces are the only resource type that has an owner, and they can only have a single address as the owner (if you need to allow multiple people to act as owner, you can configure a 1/n multi-sig as the owner, and then any of them can do the owner only actions).
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:55]:
```suggestion
`Namespace` is the only resource type that has an owner.
The owner is a single account (but by using multi-sig accounts ownership can be shared).
```

File: next-docs/pages/world/namespaces-access-control.mdx [hunk 15]
```diff
@@ -1 +1,74 @@
...
+- Register new [tables](/store/tables).
+- Register [store hooks](/store/store-hooks) for tables in the namespace.
+- Register new [systems](/world/systems).
+- Upgrade existing systems.
+- Register [system hooks](/world/system-hooks) for systems in the namespace.
+- Register [function selectors](/world/function-selectors) for systems in the namespace.
+
+Owners can modify which contract a `World` uses for each `System`, so a namespace with an owner is not truly decentralized, unless the owner is some kind of distributed governance mechanism.
+When the application is ready to be truly decentralized, the owner can revoke previously granted access as needed and then transfer ownership to `address(0)`.
+Once this is done no new access can be granted, and ownership cannot be transferred again.
```
Comment (alvrs) [inline next-docs/pages/world/namespaces-access-control.mdx:74]: "True decentralization" is nuanced and can be argued about, I'd simplify this paragraph to:

```suggestion
Note: once ownership of a namespace is burned (i.e. transferred to `address(0)`), no new access can be granted or revoked and ownership can't be transferred again.
```

File: next-docs/pages/world/namespaces-access-control.mdx (hunk 8 above)
Comment (qbzzt) [reply next-docs/pages/world/namespaces-access-control.mdx:37]: The secret information isn't *on the blockchain*. But I'll make it clearer.

**Review by alvrs — APPROVED** [review]

**Pull request merged**